packages/@lwc/synthetic-shadow/src/polyfills/event/polyfill.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { defineProperties, isNull, isUndefined } from '@lwc/shared';
+import { pathComposer } from '../../3rdparty/polymer/path-composer';
+import { retarget } from '../../3rdparty/polymer/retarget';
+import { eventTargetGetter, eventCurrentTargetGetter } from '../../env/dom';
+import { eventsDispatchedDirectlyOnShadowRoot } from '../../faux-shadow/events';
+import { getShadowRoot, isHostElement } from '../../faux-shadow/shadow-root';
+import { EventListenerContext, eventToContextMap } from '../../faux-shadow/events';
+import { getNodeOwnerKey } from '../../shared/node-ownership';
+import { getOwnerDocument } from '../../shared/utils';
+
+export function targetGetter(this: Event): EventTarget | null {
+    const originalCurrentTarget = eventCurrentTargetGetter.call(this);
+    const originalTarget = eventTargetGetter.call(this);
+    const composedPath = pathComposer(originalTarget, this.composed);
+    const doc = getOwnerDocument(originalTarget as Node);
```

Comment:
There is no guarantee that the `originalTarget` is a `Node`. 

```js
// When the event is not dispatched
new Event('foo').target // null

// When dispatched on an EventTarget that isn't a Node.
window.addEventListener('foo', evt => evt.target) // window
window.dispatchEvent(new Event('foo'));
```

We should assert first that `originalTarget` is a `Node` and remove all the type guards here.